lms/djangoapps/commerce/views.py

```diff
@@ -0,0 +1,179 @@
+
+    def _enroll(self, course_key, user):
+        """ Enroll the user in the course. """
+        CourseEnrollment.enroll(user, course_key, True)
+
+    def post(self, request, *args, **kwargs):  # pylint: disable=unused-argument
+        """
+        Attempt to purchase the course seat and enroll the user.
+        """
+        user = request.user
+        valid, course_key, error = self._is_data_valid(request)
+        if not valid:
+            return DetailResponse(error, status=HTTP_406_NOT_ACCEPTABLE)
+
+        # Ensure that the E-Commerce API is setup properly
+        ecommerce_api_url = getattr(settings, 'ECOMMERCE_API_URL')
+        ecommerce_api_signing_key = getattr(settings, 'ECOMMERCE_API_SIGNING_KEY')
+
+        if not (ecommerce_api_url and ecommerce_api_signing_key):
+            self._enroll(course_key, user)
+            msg = 'E-Commerce API not setup. Enrolled {} in {} directly.'.format(user.username, unicode(course_key))
+            log.debug(msg)
+            return DetailResponse(msg)
+
+        # Default to honor mode. In the future we may expand this view to support additional modes.
+        mode = CourseMode.DEFAULT_MODE_SLUG
+        course_modes = CourseMode.objects.filter(course_id=course_key, mode_slug=mode, sku__isnull=False)
+
+        # If there are no course modes with SKUs, enroll the user without contacting the external API.
+        if not course_modes.exists():
+            msg = _('The %(enrollment_mode)s mode for %(course_id)s does not have a SKU. '
+                    'Enrolling %(username)s directly.')
+            msg = msg % {
+                'enrollment_mode': mode,
+                'course_id': unicode(course_key),
+                'username': user.username
+            }
+            log.debug(msg)
+            self._enroll(course_key, user)
+            return DetailResponse(msg)
+
+        # Contact external API
+        headers = {
+            'Content-Type': 'application/json',
+            'Authorization': 'JWT {}'.format(self._get_jwt(user))
+        }
+
+        url = '{}/order'.format(ecommerce_api_url.strip('/'))
+
+        try:
+            response = requests.post(url, data={'sku': course_modes[0].sku}, headers=headers)
+        except Exception as ex:  # pylint: disable=broad-except
+            log.exception('Call to E-Commerce API failed: %s.', ex.message)
+            return ApiErrorResponse()
+
+        status_code = response.status_code
+
+        try:
+            data = response.json()
+        except JSONDecodeError as ex:
+            log.error('E-Commerce API response is not valid JSON.')
+            return ApiErrorResponse()
+
+        if status_code == HTTP_200_OK:
+            order_number = data.get('number')
+            order_status = data.get('status')
+            if order_status == OrderStatus.COMPLETE:
+                msg = _('Order %(order_number)s was completed.') % {'order_number': order_number}
+                log.debug(msg)
```

Comment:
Why not use the logger's built-in formatting?
```python
log.debug(u'Order %s was completed.', order_number)
```
I don't think log messages need to be marked for translation, either.